faiss/impl/ResultCollector.h

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <unordered_set>
+#include <vector>
+
+#include <faiss/MetricType.h>
+#include <faiss/utils/Heap.h>
+
+/**
+ * ResultCollector is intended to define how to collect search result
+ * For each single search result, collect method will be called.
+ * After every results are collected, post_process method is called at the end.
+ */
+
+namespace faiss {
+
+/** Encapsulates a set of ids to handle. */
+struct ResultCollector {
+    /**
+     * For each result, collect method is called to store result
+     * @param k         number of vectors to search
+     * @param nres      number of results in queue
```

Comment:
Why is this passed by ref?

Reply:
nres need to be updated inside the method.